OneSignalSDK/onesignal/src/main/java/com/onesignal/HmsMessageServiceOneSignal.java

```diff
@@ -20,15 +22,18 @@
  */
 public class HmsMessageServiceOneSignal extends HmsMessageService {
 
+
     /**
      * When an app calls the getToken method to apply for a token from the server,
      * if the server does not return the token during current method calling, the server can return the token through this method later.
      * This method callback must be completed in 10 seconds. Otherwise, you need to start a new Job for callback processing.
+     *
      * @param token token
+     * @param bundle bundle
      */
     @Override
-    public void onNewToken(String token) {
-        OneSignalHmsEventBridge.onNewToken(this, token);
+    public void onNewToken(String token, Bundle bundle) {

```

Comment:
@jkasten2 There won't be a problem because they will have implemented the 

 public void onNewToken(String token) that will be calling the 
 
```  @Deprecated
    public static void onNewToken(@NonNull Context context, @NonNull String token) {
        onNewToken(context, token, null);
    }```
    
    
The new method won't be called unless the user overrides it. But we should recommend updating

Reply:
The older `hms:push` doesn't know about the new `onNewToken(String, Bundle)`, only `onNewToken(String)`. If the app includes our new OneSignal-Android-SDK (with this change) AND the app has the old `hms:push` version 4 I believe HMS is going to try to call `onNewToken(String)` on our `HmsMessageServiceOneSignal` and it's not going to find it. It think it might crash, or OneSignal won't get a push token.

If the app developer has their own class that extends `HmsMessageService` this won't be an issue, as `HmsMessageServiceOneSignal` won't be called by HMS. We instead instruct the app developer to call `OneSignalHmsEventBridge` from their class which I see you have kept both methods for backwards compatibility as expected.